src/transform-sdk/rust/core-sys/src/lib.rs

```diff
@@ -179,6 +179,7 @@ where
             },
             writer,
         )
-        .expect("transforming record failed");
+        .inspect_err(|e| eprintln!("transforming record failed with error: {:?}", e))
+        .unwrap_or(())
```

Comment:
Looks good, and thanks for the change. I'm hesitant to merge right away since the panic-on-expect behavior feels intentional, and it may be directly related to the transform execution model viz how we calculate backoffs and such. Just need to give it some though.

Reply:
This will drop records that fail on the floor, which is not a safe default because there is no way to know what failed where. I don't agree with this change (of course my opinion doesn't matter anymore), users should opt into this. For consistency other SDKs should change to this (go sdk panics on failures).

This default pauses until a deploy happens and the developers fix the code.

Reply:
@rockwotj I don’t see the pause behavior you describe. What I see is the same record being attempted to be reprocessed repeatedly, causing the same panic over and over again spamming the logs.

If the behavior is to panic, there’s also no reason to use a `Result` type. Just have the transform code panic directly. Returning `Err` just to cause a panic is very surprising as someone writing code against the api.

Reply:
Yes there are retred because we can't tell the difference between an OOM where a retry can fix and a permanent failure right now. That is fixable without SDK changes (basically if the user code fails as the first batch it will never succeed)

RE API - it's future proofing it. We should have an SDK or yanl configurable way to set the dead letter queue, then we can do the dead letter queue pattern automatically for a user.